02.calendar/cal.rb

```diff
@@ -0,0 +1,48 @@
+require "date"
+require "optparse"
+
+opt = OptionParser.new
+opt.on("-y [VAL]") {|year| @year = year}
+opt.on("-m [VAL]") {|month| @month = month}
+opt.parse(ARGV)
+
+if @year != nil
+  @year
+else
+  @year = Date.today.year
+end
+if @month != nil
+  @month
+else
+  @month = Date.today.month
+end
+
+puts "#{@month}月 #{@year}".center(20)
+
+dayweeks = ["日","月","火","水","木","金","土"]
+dayweeks.each do |dayweek|
+  print "#{dayweek} "
+end
+puts "\n"
+
+first = Date.new(@year.to_i,@month.to_i,1)
+last = Date.new(@year.to_i,@month.to_i,-1)
+days = [*(first..last)].map {|day| day.day}
+weeks = [*(first..last)].map {|week| week.wday}
+
+day_week_hash = Hash[days.zip weeks]
+
+n = Date.new(@year.to_i,@month.to_i,1).wday
+print "   " * n
+
+day_week_hash.each do |kye,value|
```

Comment:
`kye` じゃなくて `kye` でしょうか？

Reply:
```suggestion
day_week_hash.each do |day, wday|
```

展開されるハッシュを key, value という変数にいれると抽象的すぎるので
`day, wday` のようにもう少し具体的な変数名にするとわかりやすいかもしれませんね。